components/ShakeComponent.js

```diff
@@ -0,0 +1,242 @@
+
+    const [data, setData] = useState({
+        x: 0,
+        y: 0,
+        z: 0,
+    })
+
+    useEffect(() => {
+        fetchData()
+    }, [])
+
+    useEffect(() => {
+        subscribe()
+        return () => unsubscribe()
+    }, [])
+
+    useEffect(() => {
+        if (isShaking(data)) {
+            fetchData()
+        }
+    }, [data])
+
+    const subscribe = () => {
+        setSubscription(
+            Accelerometer.addListener((accelerometerData) => {
+                setData(accelerometerData)
+            })
+        )
+    }
+
+    const unsubscribe = () => {
+        subscription && subscription.remove()
+        setSubscription(null)
+    }
+
+    const fetchData = () => {
+        setLoading(true)
+        fetch("https://www.boredapi.com/api/activity")
+            .then(res => res.json())
+            .then(data => setTimeout(() => setFetchedData(data), 600))
+            .finally(() => setTimeout(() => setLoading(false), 600))
+            Vibration.vibrate()
+    }
+
+    const changeImage = () => {
+        switch (fetchedData.type) {
+            case "social":
+                return <Image source={require("../assets/social.png")} 
+                        style={{ resizeMode: "contain", height: 280, width: 280 }}  />
+                break
+            case "cooking":
+                return <Image source={require("../assets/cooking.png")} 
+                            style={{ resizeMode: "contain", height: 280, width: 280 }} />
+                break
+            case "busywork":
+                return <Image source={require("../assets/busywork.png")} 
+                        style={{ resizeMode: "contain", height: 280, width: 280 }} />
+                break
+            case "education":
+                return <Image source={require("../assets/education.png")} 
+                        style={{ resizeMode: "contain", height: 280, width: 280 }} />
+                break
+            case "diy":
+                return <Image source={require("../assets/diy.png")} 
+                        style={{ resizeMode: "contain", height: 280, width: 280 }} />
+                break
+            case "recreational":
+                return <Image source={require("../assets/recreational.png")} 
+                        style={{ resizeMode: "contain", height: 280, width: 280 }} />
+                break
+            case "charity":
+                return <Image source={require("../assets/charity.png")} 
+                        style={{ resizeMode: "contain", height: 280, width: 280 }} />
+                break
+            case "music":
+                return <Image source={require("../assets/music.png")} 
+                        style={{ resizeMode: "contain", height: 280, width: 280 }} />
+                break
+            default:
+                return <Image source={require("../assets/relaxing.png")} 
+                        style={{ resizeMode: "contain", height: 280, width: 280 }} />
+        }
```

Comment:
Another way to avoid repeating the code. You can return only the image source here:

return  `".required(./assets/relaxing.png)"`

, instead of rewriting them like this

```
<Image source={require("../assets/relaxing.png")} 
                         style={{ resizeMode: "contain", height: 280, width: 280 }} />
```

In the function in the line 164, we can replace it with : 
```
<Image source={changeImage()} 
                         style={{ resizeMode: "contain", height: 280, width: 280 }} />
                         
```

Hope you find it helpful